hazelcast-jet-kafka/src/main/java/com/hazelcast/jet/connector/kafka/ReadKafkaP.java

```diff
@@ -145,33 +155,62 @@ public void close() {
      * @param deserializeKey         function for deserializing keys
      * @param deserializeValue       function for deserializing values
      */
-    static <K, V> ProcessorMetaSupplier readKafka(String zkAddress, String groupId, String topicId,
-                                                  String brokerConnectionString,
-                                                  Function<byte[], K> deserializeKey,
-                                                  Function<byte[], V> deserializeValue) {
+    public static <K, V> ProcessorMetaSupplier readKafka(String zkAddress, String groupId, String topicId,
+                                                         String brokerConnectionString,
+                                                         Function<byte[], K> deserializeKey,
+                                                         Function<byte[], V> deserializeValue) {
+        return readKafka(zkAddress, groupId, topicId, brokerConnectionString, deserializeKey, deserializeValue,
+                CURRENT_POSITION_OFFSET_MAPPER);
+    }
+
+    /**
+     * Returns a meta-supplier of processors that consume a kafka topic and emit
+     * items from it as {@code Map.Entry} instances.
+     *
+     * <p>
+     *     You can specify a partition to offset mapper to mark the start offset of each partition.
+     *     Any negative value as an offset will throw {@code IllegalArgumentException}
+     * </p>
+     *
+     * @param <K>                    type of keys read
+     * @param <V>                    type of values read
+     * @param zkAddress              zookeeper address
+     * @param groupId                kafka consumer group name
+     * @param topicId                kafka topic name
+     * @param brokerConnectionString kafka broker address
+     * @param deserializeKey         function for deserializing keys
+     * @param deserializeValue       function for deserializing values
+     * @param partitionOffsetMapper  function for mapping partitionOffsets
+     */
+    public static <K, V> ProcessorMetaSupplier readKafka(String zkAddress, String groupId, String topicId,
+                                                         String brokerConnectionString,
+                                                         Function<byte[], K> deserializeKey,
+                                                         Function<byte[], V> deserializeValue,
+                                                         BiFunction<Integer, Long, Long> partitionOffsetMapper) {
```

Comment:
the list of parameters is getting really long now. Maybe we should just accept the Properties object?

This seems to be the typical case when creating KafkaConsumer: https://www.confluent.io/blog/tutorial-getting-started-with-the-new-apache-kafka-0-9-consumer-client/

Also it has support for some custom deserializer, from the `Deserializer` interface.

Could you do some more research around in this area?